tests/e2e/plugins/index.js

```diff
@@ -5,20 +5,48 @@
 // as explained in the cypress docs
 // https://docs.cypress.io/api/plugins/preprocessors-api.html#Examples
 
-/* eslint-disable global-require, arrow-body-style */
-// const webpack = require('@cypress/webpack-preprocessor')
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const webpack = require('@cypress/webpack-preprocessor');
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const vueCliOptions = require('@vue/cli-service/webpack.config');
+
+/**
+ * We must use a different webpack congiuration for cypress. This is because
+ * we need to use a different tsconfig.json file to compile the .ts files.
+ *
+ * Here we aim to use the following webpack rules from vue cli:
+ * - resolve (just alias, extensions, and modules)
+ * - resolveLoader
+ * - The rule for .ts files
+ *
+ * https://github.com/cypress-io/cypress-webpack-preprocessor
+ * Use `vue inspect` to see the webpack configuration for the app
+ * Use `vue inspect --rule ts` to see the .ts file rule
+ */
+const options = webpack.defaultOptions;
```

Comment:
id recommend doing this in a separate webpack.config.js file. also, have you looked at webpack-merge? could help here

Reply:
I'm following the recommendation on how to modify default values from the readme: https://github.com/cypress-io/cypress-webpack-preprocessor#modifying-default-options

webpack-merge might help... But ultimately I'm only looking to copy specific rules from the Vue CLI webpack config. If I were to copy the entire webpack config, the cypress tests do not compile correctly, and I was unable to figure out what rule was the culprit. Here I cherry pick the config for `resolve`, `resolveLoader`, and the rule for `.ts` files.

Reply:
that's fine; i'm thinking just doing something like

```js
// webpack.config.js
const { defaultOptions } = require('@cypress/webpack-preprocessor')
const vueConfig = require('@vue/cli-service/webpack.config')
const _ = require('lodash')
const webpackMerge = require('webpack-merge')

const tsRule = // you can keep your original code here

export default webpackMerge(
  defaultOptions,
  _.at(vueConfig, [
    'resolve.alias',
    'resolve.extensions',
    'resolve.modules',
    'resolveLoader',
  ]),
  {
    module: {
      rules: [tsRule],
    },
  },
)
```

for one, I dislike modifying objects (only pure operations!! lol) but also the use of lodash helps make things a bit easier to understand

also, you could look into [`immer`](https://immerjs.github.io/immer/docs/introduction), which might help with `Show` modification stuff. I like it because you can still do javascript modification things (which is the way JS wants you to do things) but it doesn't mutate anything, keeping states kosher

Reply:
:+1: Will take a look at this. Thank you for the suggestions!